Clarify difference between Cadence type and TypeValue (#39)
fxamacker/ccf_draft: ccf_specs.md

For example, Cadence `UInt8` is encoded as CBOR positive integer, Cadence `String` is encoded as CBOR text string,
Cadence `Address` is encoded as CBOR byte string, and Cadence struct data is encoded as an array of its raw field data.

> ### Cadence Type and TypeValue

Reviewer: ```suggestion
### Cadence Types and Type Values
```

> Cadence types and Cadence `TypeValue` are encoded differently.  They contain different data because they are used differently.

Reviewer: ```suggestion
Cadence types and Cadence type values (run-time types) are encoded differently.  They contain different data because they are used differently.
```

Author: @turbolent 

> Cadence types and Cadence type values (run-time types) are encoded differently.

Can Cadence type values be static types as well as run-time types?

Reviewer: Cadence type values are also called "run-time types", and they are static types reified as a value

Cadence types are used to decode Cadence data, so they only contain information needed for decoding.  For example, composite type's field info is needed to decode composite value.  However, interface type's field info isn't needed to decode values implementing interface type.

> Cadence `TypeValue` is a Cadence value which provides comprehensive information about a type.  For example, composite type value and interface type value contain info about both fields and initializers.

Reviewer: ```suggestion
Cadence type value is a Cadence value which provides comprehensive information about a type.  For example, composite type value and interface type value contain info about both fields and initializers.
```
